lib/js/components/Toggles/ToggleButton/ToggleButton.vue

```diff
@@ -103,236 +240,318 @@ $toggle-button-colors: (
 	display: flex;
 	justify-content: center;
-	min-height: $toggle-button-size-medium;
-	min-width: $toggle-button-size-medium;
 	outline: 6px solid transparent;
-	// vertical passing is smaller than designed, but we compensate that with `min-height`.
-	// This shortcut allows us to use the same padding on .filled and .outlined versions, so number of variants is cut by half
-	padding: $space-xxs;
 	pointer-events: none;
 	transition: color ease-in-out $default-transition-time,
 		border-color ease-in-out $default-transition-time,
 		background-color ease-in-out $default-transition-time,
 		outline-color ease-in-out $default-transition-time;
 
+	&__contentWrapper {
+		align-items: center;
+		display: flex;
+		gap: $space-xxxxs;
+		justify-content: center;
+	}
+
 	&__content {
+		@include label-l-default-bold;
+
 		overflow: hidden;
 		text-overflow: ellipsis;
 		white-space: nowrap;
-	}
-
-	&.-long {
-		padding: $space-xxs $space-s;
 
-		&.-rounded {
-			padding: $space-xxs $space-xs;
+		&.-uppercase {
+			@include label-l-extensive-bold-uppercase;
 		}
 	}
 
 	&.-rounded {
 		border-radius: $radius-s;
 	}
 
-	&.-outlined {
-		border: 2px solid;
-	}
-
-	&.-selected {
-		color: $color-inverted-text;
-	}
-
 	&.-interactive {
 		cursor: pointer;
 		pointer-events: initial;
 	}
 
-	&.-xSmall {
-		min-height: $toggle-button-size-x-small;
-		min-width: $toggle-button-size-x-small;
-		padding: $space-xxxxs;
-
-		&.-long {
-			padding: $space-xxxxs $space-xs;
-
-			&.-rounded {
-				padding: $space-xxxxs $space-xxs;
-			}
+	&.-loading {
+		#{$root}__contentWrapper {
+			opacity: 0;
 		}
 	}
 
 	&.-small {
 		min-height: $toggle-button-size-small;
 		min-width: $toggle-button-size-small;
-		padding: $space-xxxs;
+		padding: substract-border($space-xxxs, 'small') substract-border($space-xs, 'small');
 
-		&.-long {
-			// Exception in padding rules in Figma. -capsule has the same padding as -rounded
-			padding: $space-xxxs $space-xs;
+		&.-hasSmallHorizontalPadding {
+			padding: substract-border($space-xxxs, 'small') substract-border($space-xxxxxs, 'small');
+		}
+
+		#{$root}__content.-small {
+			@include label-s-default-bold;
 
-			&.-rounded {
-				padding: $space-xxxs $space-xs;
+			&.-uppercase {
+				@include label-s-extensive-bold-uppercase;
 			}
 		}
 	}
 
-	&.-elevationXSmall,
-	&.-elevationSmall {
-		background-color: $color-default-background;
+	&.-medium {
+		min-height: $toggle-button-size-medium;
+		min-width: $toggle-button-size-medium;
+		padding: substract-border($space-xxs, 'medium') substract-border($space-xs, 'medium');
+
+		&.-hasSmallHorizontalPadding {
+			padding: substract-border($space-xxs, 'medium')
+				substract-border($space-xxxxxs, 'medium');
+		}
+
+		#{$root}__content.-small {
+			@include label-m-default-bold;
+
+			&.-uppercase {
+				@include label-m-extensive-bold-uppercase;
+			}
+		}
 	}
 
-	&.-elevationSmall {
-		box-shadow: $shadow-l;
+	&.-large {
+		border: $toggle-button-border-size-large solid;
```

Comment:
this shorthand sets `border-color: initial`. Because of that we have wrong color in `L` size

Reply:
I believe the color was correct but overrides in DevTools could be confusing, so I split it: [`c492df2` (#303)](https://github.com/bethinkpl/design-system/pull/303/commits/c492df26af63c0682c87de88dcc7492908847c7b)

Reply:
![Zaznaczenie_135](https://github.com/bethinkpl/design-system/assets/2725894/1d2ea2eb-5c4b-4567-8ee2-79ed87f88863)
![Zaznaczenie_136](https://github.com/bethinkpl/design-system/assets/2725894/3aec1bae-17dc-4db0-a6cc-49588e6f973c)
Am I color blind?

Reply:
It turns out I am. 🙂 Anyway, it's correct after the split.

Reply:
https://developer.mozilla.org/en-US/docs/Web/CSS/border
> color: Sets the color of the border. Defaults to currentcolor if absent.